content/introduction/how-ipfs-works.md

```diff
@@ -0,0 +1,82 @@
+
+
+### 3 \\ The DHT
+
+<!-- <img alt="IPFS Stack - Routing and Exchange" src="../assets/ipfs_stack-exchange_routing.png" width="200px" /> -->
+
+So, to recap, IPFS lets you to give CIDs to content, and link that content together in a Merkle-DAG using IPLD. Now let’s move on to the last piece: how you find and move that content around!
+
+To find which peers are hosting the content you’re after (_discovery_), IPFS uses a [_distributed hash table_](https://en.wikipedia.org/wiki/Distributed_hash_table), or DHT. A hash table is a database of keys to values. A _distributed_ hash table is one where the table is split across all the peers in a distributed network. To find content, you ask these peers.
+
+The <a hrefm src="https://libp2p.io/">libp2p project</a> is the part of the IPFS ecosystem that provides the DHT and handles peers connecting and talking to each other. It can be used as a tool for other distributed systems as well; just as there is “IPFS-flavored IPLD,” there’s also “IPFS-flavored libp2p.”
+
+Once you know where your content is (ie, which peer or peers are storing the various blocks that make up the content you’re after), you use the DHT **again** to find the current location of those peers (_routing_). So, in order to get to content, you use libp2p to query the DHT twice.
+
+**TODO: —> this description of exchange might not be right, and we perhaps need something about bootstrapping, Bitswap, or Graphsync.** You’ve discovered your content, and you’ve found the current location(s) of that content — now you need to connect to that content (_exchange_). libp2p again to the rescue! _[Maybe something here about bootstrapping, Bitswap, or Graphsync.]_
+
+What makes the IPFS protocol especially useful for peer to peer connections is _connection multiplexing_. Traditionally, every service in a system would open a different connection to remotely communicate with other services of the same kind. Using IPFS, you open just one connection, and you multiplex everything on that. For everything your peers have to talk to each other about, you send a little bit of each thing, and the other end knows how to sort those chunks where they belong.
+
+This is useful because establishing connections is usually hard to set up and expensive to maintain. With multiplexing, once you have that connectivity, you can do whatever you need on it.
```

Comment:
```suggestion
This is useful because establishing connections is usually hard to set up and expensive to maintain. With multiplexing, once you have that connection, you can do whatever you need on it.
```

Reply:
Fixed in b4f174e